src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/configuration/RabbitMQConfiguration.kt

```diff
@@ -37,12 +38,60 @@ data class ConnectionManagerConfiguration(
     var minConnectionRecoveryTimeout: Int = 10000,
     var maxConnectionRecoveryTimeout: Int = 60000,
     val prefetchCount: Int = 10,
+    var retryTimeDeviationPercent: Int = 10,
     val messageRecursionLimit: Int = 100,
     val workingThreads: Int = 1,
-    val confirmationTimeout: Duration = Duration.ofMinutes(5)
+    val confirmationTimeout: Duration = Duration.ofMinutes(5),
+    val isAutomaticRecoveryEnabled: Boolean = true
 ) : Configuration() {
     init {
         check(workingThreads > 0) { "expected 'workingThreads' greater than 0 but was $workingThreads" }
         check(!confirmationTimeout.run { isNegative || isZero }) { "expected 'confirmationTimeout' greater than 0 but was $confirmationTimeout" }
     }
+
+    fun createRetryingDelaySequence(): Sequence<RetryingDelay> {
+        return generateSequence(RetryingDelay(0, minConnectionRecoveryTimeout)) {
+            RetryingDelay(it.tryNumber + 1, RetryingDelay.getRecoveryDelay(
+                it.tryNumber + 1,
+                minConnectionRecoveryTimeout,
+                maxConnectionRecoveryTimeout,
+                maxRecoveryAttempts,
+                retryTimeDeviationPercent
+            ))
+        }
+    }
+
+}
+
+data class RetryingDelay(val tryNumber: Int, val delay: Int) {
+    companion object {
+        @JvmStatic
+        fun getRecoveryDelay(
+            numberOfTries: Int,
+            minTime: Int,
+            maxTime: Int,
+            maxRecoveryAttempts: Int,
+            deviationPercent: Int
+        ): Int {
+            return if (numberOfTries <= maxRecoveryAttempts) {
+                getRecoveryDelayWithIncrement(numberOfTries, minTime, maxTime, maxRecoveryAttempts)
+            } else getRecoveryDelayWithDeviation(maxTime, deviationPercent)
```

Comment:
Please, add brackets to the `else` block as well. It makes the whole statement more readable and consistent